react/SelectBox/SelectBox.jsx

```diff
@@ -100,13 +103,63 @@ const CheckboxOption = ({ ...props }) => <Option {...props} withCheckbox />
 
 CheckboxOption.propTypes = {}
 
-const SelectBox = ({ components, styles, ...props }) => (
-  <ReactSelect
-    components={{ Option, ...components }}
-    styles={{ ...customStyles, ...styles }}
-    {...props}
-  />
-)
+class SelectBox extends Component {
+  state = {
+    isOverlayVisible: false
+  }
+
+  hideOverlay() {
+    this.setState({ isOverlayVisible: false })
+  }
+
+  onBlur = event => {
+    this.hideOverlay()
+
+    if (typeof this.props.onChange === 'function') {
+      this.props.onChange(event)
+    }
+  }
+
+  onChange = event => {
+    this.hideOverlay()
+
+    if (typeof this.props.onChange === 'function') {
+      this.props.onChange(event)
+    }
+  }
+
+  onFocus = event => {
+    this.setState({ isOverlayVisible: true })
+
+    if (typeof this.props.onFocus === 'function') {
+      this.props.onFocus(event)
+    }
+  }
+
+  render() {
+    const {
+      components,
+      styles,
+      breakpoints: { isMobile },
+      ...props
+    } = this.props
+    const showOverlay = this.state.isOverlayVisible && isMobile
+
+    return (
+      <div>
```

Comment:
The wrapping in a `div` can change the display of the resulting component from `inline-block` (if it is its default value at the moment) to `block`, is it OK ?

Reply:
I had to add a `className` prop to `<SelectBox />` and apply it to this div. It may be a breaking change if the app is passing style to the `<SelectBox/>` root element via react-select's styles API. It now have to pass it via `className`.